src/segment-loader.js

```diff
@@ -2317,8 +2317,8 @@ export default class SegmentLoader extends videojs.EventTarget {
     // Although transmuxing is done, appends may not yet be finished. Throw a marker
     // on each queue this loader is responsible for to ensure that the appends are
     // complete.
-    const waitForVideo = this.loaderType_ === 'main' && this.startingMedia_.hasVideo;
-    const waitForAudio = !this.audioDisabled_ && this.startingMedia_.hasAudio;
+    const waitForVideo = this.loaderType_ === 'main' && this.startingMedia_ && this.startingMedia_.hasVideo;
```

Comment:
Is there a case where we're trying to perform this operation (and the check in updateTimingInfoEnd) without having startingMedia?

Reply:
I only really see it when we are going to player error in some way, or when we try to load something that we shouldn't be, like loading a webm in the muxer.

Reply:
Would it be better to fail quickly with an exception there? Since it can otherwise lead to undefined behavior, and will help to surface issues where we might be trying to do something that we shouldn't be doing.

Reply:
OK I triggered an error for this case, probably a good idea to blacklist such a playlist and try other in this case too.